code/controllers/subsystem/vote.dm

```diff
@@ -234,97 +231,74 @@ SUBSYSTEM_DEF(vote)
 		return TRUE
 	return FALSE
 
-/datum/controller/subsystem/vote/proc/interface(client/C)
-	if(!C)
+/mob/verb/vote()
+	set category = "OOC"
+	set name = "Vote"
+	SSvote.ui_interact(usr)
+
+/datum/controller/subsystem/vote/ui_state()
+	return GLOB.always_state
+
+/datum/controller/subsystem/vote/ui_interact(mob/user, datum/tgui/ui)
+	// Tracks who is voting
+	if(!(user.client?.ckey in voting))
+		voting += user.client?.ckey
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Vote")
+		ui.open()
+
+/datum/controller/subsystem/vote/ui_data(mob/user)
+	var/list/data = list(
+		"mode" = mode,
+		"voted" = voted,
+		"voting" = voting,
+		"choices" = list(),
+		"question" = question,
+		"initiator" = initiator,
+		"started_time" = started_time,
+		"time_remaining" = time_remaining,
+		"lower_admin" = !!user.client?.holder,
+		"generated_actions" = generated_actions,
+		"avm" = CONFIG_GET(flag/allow_vote_mode),
+		"avmap" = CONFIG_GET(flag/allow_vote_map),
+		"avr" = CONFIG_GET(flag/allow_vote_restart),
+		"selectedChoice" = choice_by_ckey[user.client?.ckey],
+		"upper_admin" = check_rights_for(user.client, R_ADMIN),
+	)
+
+	for(var/key in choices)
```

Comment:
You're sending this full data to every client, so anyone can see other votes, you can't be verifying admin rights client-side.

Reply:
testing a solution to this now